common/common/src/main/java/io/helidon/common/Weights.java

```diff
@@ -0,0 +1,153 @@
+        if (null != weightAnnot) {
+            return weightAnnot.value();
+        }
+
+        return defaultWeight;
+    }
+
+    /**
+     * Find weight for an instance.
+     * First checks if instance is {@link io.helidon.common.Weighted}. If so,
+     * uses the value from it.
+     * Then checks for {@link Weight} annotation.
+     * If none of the above is found, returns the default weight.
+     *
+     * @param anObject      object to find weight for
+     * @param defaultWeight default weight to use
+     * @return weight of the object or default provided
+     */
+    public static double find(Object anObject, double defaultWeight) {
+        if (anObject instanceof Class) {
+            return find((Class<?>) anObject, defaultWeight);
+        }
+        if (anObject instanceof Weighted) {
+            return ((Weighted) anObject).weight();
+        }
+        Weight weight = null;
+        // first go through super classes
+        Class<?> current = anObject.getClass();
+        while (weight == null) {
+            weight = current.getAnnotation(Weight.class);
+            current = current.getSuperclass();
+            if (current.equals(Object.class)) {
+                break;
+            }
+        }
+        if (weight == null) {
+            // then through interfaces (only first level)
+            Class<?>[] interfaces = anObject.getClass().getInterfaces();
+            for (Class<?> anInterface : interfaces) {
+                weight = anInterface.getAnnotation(Weight.class);
+                if (weight != null) {
+                    break;
+                }
+            }
+        }
+
+        if (weight == null) {
+            return defaultWeight;
+        }
+        return weight.value();
+    }
+
+    /**
+     * Sort the list based on weights.
+     * <ul>
+     * <li>If element implements {@link io.helidon.common.Weighted}, uses its weight.</li>
+     * <li>If element is a class and has annotation {@link Weight}, uses its weight</li>
+     * <li>If element is any object and its class has annotation {@link Weight}, uses its weight</li>
+     * </ul>
+     *
+     * @param list list to sort
+     */
+    public static void sort(List<?> list) {
+        list.sort(weightComparator());
+    }
+
+    /**
+     * Returns a comparator for two objects, the classes for which are implementations of
+     * {@link io.helidon.common.Weighted}, and/or optionally annotated with {@link Weight}
+     * and which applies a specified default weight if either or both classes lack the annotation.
+     *
+     * @param <S> type of object being compared
+     * @return comparator
+     */
+    public static <S> Comparator<S> weightComparator() {
+        return (o1, o2) -> {
+            if (o1 == null) {
+                return o2 == null ? 0 : 1;
+            }
+            if (o2 == null) {
+                return -1;
+            }
+
+            double firstWeight = find(o1, -101);
```

Comment:
Some constant for magic number would be nicer

Reply:
Using a constant, added a comment